hazelcast/src/main/java/com/hazelcast/topic/impl/TopicProxy.java

```diff
@@ -64,6 +79,33 @@ public boolean removeMessageListener(@Nonnull UUID registrationId) {
     public LocalTopicStats getLocalTopicStats() {
         return getLocalTopicStatsInternal();
     }
+
+    @Override
+    public void publishAll(@Nonnull Collection<? extends E> messages) {
+        checkNotNull(messages, NULL_MESSAGE_IS_NOT_ALLOWED);
+        checkNoNullInside(messages, NULL_MESSAGE_IS_NOT_ALLOWED);
+        messages.forEach(this::publishInternal);
+    }
+
+    @Override
+    public CompletionStage<Void> publishAllAsync(@Nonnull Collection<? extends E> messages) {
+        checkNotNull(messages, NULL_MESSAGE_IS_NOT_ALLOWED);
+        checkNoNullInside(messages, NULL_MESSAGE_IS_NOT_ALLOWED);
+        publishAll(messages);
+        return InternalCompletableFuture.completedFuture(null);
+    }
+
+
+    protected Data[] toDataArray(Collection<? extends E> collection) {
+        Data[] items = new Data[collection.size()];
```

Comment:
If you want, replace this with:
```java
        return collection.stream().map(item -> {
            checkNotNull(item, "collection can't contains null items");
            return toData(item);
        }).toArray(Data[]::new);
```

Reply:
Much more beautiful. :) 
I've changed it.